lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/REEFUncaughtExceptionHandler.java

```diff
@@ -51,19 +51,20 @@ public REEFUncaughtExceptionHandler(final Configuration errorHandlerConfig) {
 
   @Override
   public synchronized void uncaughtException(final Thread thread, final Throwable throwable) {
+
+    final String msg = "Thread " + thread.getName() + " threw an uncaught exception.";
+    LOG.log(Level.SEVERE, msg, throwable);
+
     if (this.errorHandler == null) {
       try {
-        this.errorHandler = Tang.Factory.getTang().newInjector(this.errorHandlerConfig)
-            .getInstance(REEFErrorHandler.class);
-      } catch (InjectionException ie) {
+        this.errorHandler = Tang.Factory.getTang()
+            .newInjector(this.errorHandlerConfig).getInstance(REEFErrorHandler.class);
```

Comment:
Good point! I think we should use the same error handler as in `REEFEnvironment`. The problem was in the original code, though, and I am not sure what is the most elegant way to solve it. I will open a new JIRA to fix it later.